fastlane/lib/fastlane/actions/delete_keychain.rb

```diff
@@ -7,9 +7,16 @@ def self.run(params)
         original = Actions.lane_context[Actions::SharedValues::ORIGINAL_DEFAULT_KEYCHAIN]
 
         search_paths = []
-        search_paths << File.expand_path(params[:name]) if params[:name]
-        search_paths << File.expand_path(File.join("~", "Library", "Keychains", params[:name])) if params[:name]
-        search_paths << File.expand_path(params[:keychain_path]) if params[:keychain_path]
+        if params[:name]
+          search_paths << File.expand_path(params[:name])
```

Comment:
We recently introduced a new [helper](https://github.com/fastlane/fastlane/blob/5eed4da1559136ff4c490029ebf0b794c826ded2/fastlane_core/lib/fastlane_core/helper.rb#L180) that handles all of this. It searches all the possible locatiosn that you are adding to the search_paths (and more) so you should be able to use that helper instead of this code.